Upgrade to version 7.2.0

PetrHeinz opened the pull request:
| Q             | A
| ------------- | ---
|Description, reason for the PR| Upgrade demoshop to Shopsys Framework 7.2.0
|New feature| Yes <!-- Do not forget to update CHANGELOG.md and possibly docs/ -->
|BC breaks| What's considered a BC break in demoshop? <!-- Do not forget to update UPGRADE.md -->
|Fixes issues| ...
|Standards and tests pass| Yes
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

- using [general upgrade instructions](https://github.com/shopsys/shopsys/blob/v7.2.0/UPGRADE.md) and [the upgrade guide for 7.2.0](https://github.com/shopsys/shopsys/blob/v7.2.0/docs/upgrade/UPGRADE-v7.2.0.md)
- demoshop doesn't use kubernetes or cloud infrastructure, those upgrades are skipped

PetrHeinz committed:
infrastructure: update the definition of the postgres service in your docker-compose.yml file to use the customized configuration (#946)

PetrHeinz committed:
infrastructure: update your nginx.conf file like this to have in nginx the same limit for file size as for php from php.ini (#947)

PetrHeinz committed:
updated all shopsys/* composer dependencies to v7.2.0

PetrHeinz committed:
upgraded other composer and npm dependencies

- via `composer update` and `npm update`

PetrHeinz committed:
automatically fixed some npm dependency vulnerabilities

- via `npm audit fix` as suggested after update
- fixed 9 of 22 vulnerabilities in 2249 scanned packages

PetrHeinz committed:
app: add TransformString::removeDriveLetterFromPath transformer for all absolute paths that could be based with drive letter file systems and used by local_filesystem service (#942)

PetrHeinz committed:
app: change all occurrences of ->will($this->returnValue(…)) into ->willReturn(…) in all your TestCase tests (#939)

PetrHeinz committed:
app: remove unused @dataProvider annotation from Tests\ShopBundle\Functional\Twig\PriceExtensionTest:checkPriceFilter method (#939)

PetrHeinz committed:
app: to be more descriptive about the error caused by active TEST environment, modify ErrorController::createUnableToResolveDomainResponse() to be explicit about overwrite_domain_url parameter (#701)

PetrHeinz committed:
config: use the standard format for redis prefixes (#928)

PetrHeinz committed:
config: build version is used in redis prefix of the twig cache and contains the environment

- avoids the usage of a wrong cache
  - eg. running test locally would create a cached version of categories with different data set

PetrHeinz committed:
config: in order to have translations extracted even from overwritten templates, update your build-dev.xml file (#931)

PetrHeinz committed:
config: use redis as cache for doctrine and framework (#930)

PetrHeinz committed:
config: prepare the configuration file for cron services in your project repository (#989)

PetrHeinz committed:
config: move database_server_version parameter to parameters_common.yml (#1001)

PetrHeinz committed:
config: add env(ELASTIC_SEARCH_INDEX_PREFIX): '' into your app/config/parameters.yml.dist and also app/config/parameters.yml (#961)

PetrHeinz committed:
app: use interchangeable product filtering (#943)

PetrHeinz committed:
app: reconfigure fm_elfinder to use main_filesystem (#932)

boris-brtan commented on src/Shopsys/ShopBundle/Controller/Front/ErrorController.php:
```diff
@@ -159,6 +163,33 @@ private function getPrettyExceptionResponse(Exception $exception)
+
+        $overwriteDomainUrl = $this->getParameter('overwrite_domain_url');
+        if ($overwriteDomainUrl !== null) {
+            $content .= sprintf(" Parameter 'overwrite_domain_url' is set to '%s'.", $overwriteDomainUrl);
```
hmmm, this was done not by the instructions, would we create reverse PR into monorepo or fix it in UPGRADE-7.2.0.md ?

PetrHeinz replied:
I couldn't follow the exact upgrade instructions as the methods were not present in the class at all. So I took this approach. In my point of view, the message is really important only when `overwrite_domain_url` is set to anything but `null`. It doesn't have a direct connection to the TEST environment...

Do we really need the demoshop to be as close to `shopsys/project-base` as possible? I tried to do the upgrade just as I would my own project built on top of Shopsys Framework.

PetrHeinz replied:
OK, I've changed it to the `shopsys/project-base` implementation in c0525d054

boris-brtan commented on app/config/packages/snc_redis.yml:
```diff
@@ -5,19 +5,19 @@ snc_redis:
             dsn: 'redis://%redis_host%'
             options:
-                prefix: '%env(REDIS_PREFIX)%%build-version%doctrine_metadata_'
+                prefix: '%env(REDIS_PREFIX)%%build-version%:cache:doctrine_metadata:'
```
```suggestion
                prefix: '%env(REDIS_PREFIX)%%build-version%:cache:doctrine:metadata:'
```

PetrHeinz replied:
This leads to a more flat and compact  structure and imo it's easier to orient in:

![image](https://user-images.githubusercontent.com/10008612/57857543-a7f12a80-77ef-11e9-9e5a-7b91f122d463.png)

boris-brtan commented on app/config/packages/snc_redis.yml:
```diff
@@ -5,19 +5,19 @@ snc_redis:
             dsn: 'redis://%redis_host%'
             options:
-                prefix: '%env(REDIS_PREFIX)%%build-version%doctrine_query_'
+                prefix: '%env(REDIS_PREFIX)%%build-version%:cache:doctrine_query:'
```
```suggestion
                prefix: '%env(REDIS_PREFIX)%%build-version%:cache:doctrine:query:'
```

boris-brtan commented on app/config/packages/snc_redis.yml:
```diff
@@ -29,11 +29,12 @@ snc_redis:
             type: 'phpredis'
             alias: 'session'
             dsn: 'redis://%redis_host%'
+            options:
```
this should not be here

PetrHeinz replied:
It's more consistent when all prefixes are configured directly on the clients imo

boris-brtan commented on app/config/packages/snc_redis.yml:
```diff
@@ -29,11 +29,12 @@ snc_redis:
+                prefix: '%env(REDIS_PREFIX)%session:'
     session:
         client: 'session'
-        prefix: '%env(REDIS_PREFIX)%session_'
```
```suggestion
        prefix: '%env(REDIS_PREFIX)%session:'
```

PetrHeinz replied:
It's more consistent when all prefixes are configured directly on the clients imo

boris-brtan commented on app/config/parameters_common.yml:
```diff
@@ -3,7 +3,7 @@ parameters:
     debug.error_handler.throw_at: -1
     locale: en
-    build-version: '0000000000000000'
+    build-version: '0000000000000000_%kernel.environment%'
```
will we create some PR in monorepo?

PetrHeinz replied:
That might make sense. Here on demoshop I ended up with cached Twig template of categories being displayed in `dev` environment, which was very confusing so I've fixed it.

boris-brtan commented on CHANGELOG.md:
```diff
@@ -95,6 +95,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
     - added extra error page when current domain cannot be resolved with explicit hint about `overwrite_domain_url` parameter (for TEST environment)
     - using standard redis separators (`:`) in prefixes
     - build version is used in redis prefix of the twig cache and contains the environment (avoids the usage of a wrong cache)
+    - translations are extracted from overwritten templates now
```
are they? won't we also extract them via phing target?

PetrHeinz replied:
Makes sense, I've extracted them in 3c469ca41

boris-brtan commented on app/config/packages/snc_redis.yml:
```diff
@@ -18,6 +18,12 @@ snc_redis:
+            alias: 'framework_annotations'
+            dsn: 'redis://%redis_host%'
+            options:
+                prefix: '%env(REDIS_PREFIX)%%build-version%:cache:framework_annotations:'
```
```suggestion
                prefix: '%env(REDIS_PREFIX)%%build-version%:cache:framework:annotations:'
```

boris-brtan commented on tests/ShopBundle/Functional/Model/Product/Search/ProductSearchExportRepositoryTest.php:
```diff
@@ -0,0 +1,75 @@
+        ];
+
+        if ($productSearchExportRepository instanceof ProductSearchExportWithFilterRepository) {
+            $structure = \array_merge($structure, [
```
this is strange how `array_keys` method doesn't need global namespace prefix but `array_merge` does

PetrHeinz replied:
We don't have any standards about using functions from the global namespace... This is a copy-pasted file according to our upgrade notes.

Maybe it's something IDE autocompleted for the original author...

boris-brtan commented on src/Shopsys/ShopBundle/Resources/config/services.yml:
```diff
@@ -182,4 +186,3 @@ services:
     Shopsys\ShopBundle\Model\AdvancedSearch\ProductAdvancedSearchConfig: ~
 
     Shopsys\FrameworkBundle\Model\AdvancedSearch\ProductAdvancedSearchConfig: '@Shopsys\ShopBundle\Model\AdvancedSearch\ProductAdvancedSearchConfig'
-
```
```suggestion

```

PetrHeinz replied:
There were two newlines at the end. It's hard to see on Github web UI

boris-brtan commented on src/Shopsys/ShopBundle/Resources/config/services.yml:
```diff
@@ -64,6 +64,10 @@ services:
 
+    Shopsys\FrameworkBundle\Model\Product\ProductOnCurrentDomainFacadeInterface: '@Shopsys\FrameworkBundle\Model\Product\ProductOnCurrentDomainElasticFacade'
+
+    Shopsys\FrameworkBundle\Model\Product\Search\Export\ProductSearchExportRepository: '@Shopsys\FrameworkBundle\Model\Product\Search\Export\ProductSearchExportWithFilterRepository'
```
:smile_cat: shouldn't be here `alias:` :smile_cat:  ???

PetrHeinz replied:
I't the same thing (see [Symfony docs on Aliasing](https://symfony.com/doc/current/service_container/alias_private.html#aliasing)) and here the shorthand looks more consistent with the rest of the services.

boris-brtan reviewed (changes requested):
Hi @PetrHeinz ,
tHX for the upgrade PR, i just have a few questions and suggestions

PetrHeinz committed:
dump translations

- msgids from overwritten templates were extracted
- czech translations used from message.cs.po in shopsys/framework
